src/Swashbuckle.AspNetCore.Examples/Examples/ExamplesOperationFilter.cs

```diff
@@ -37,30 +38,50 @@ private static void SetRequestModelExamples(Operation operation, ISchemaRegistry
             foreach (var attr in actionAttributes)
             {
                 var schema = schemaRegistry.GetOrRegister(attr.RequestType);
-
+                
                 var bodyParameters = operation.Parameters.Where(p => p.In == "body").Cast<BodyParameter>();
                 var request = bodyParameters.FirstOrDefault(p => p.Schema.Ref == schema.Ref || p.Schema.Items.Ref == schema.Ref);
 
                 if (request != null)
                 {
                     var provider = ExamplesProvider(_services, attr.ExamplesProviderType);
-
+                    string name = null;
                     // var name = attr.RequestType.Name; // this doesn't work for generic types, so need to to schema.ref split
                     var parts = schema.Ref?.Split('/');
-                    if (parts == null)
+                    
+                    if (parts != null)
                     {
-                        continue;
+                        name = parts.Last();
                     }
+                    else
+                    {
+                        // schema.Ref can be null for some types, so we have to try get it by attr.RequestType.Name
+                        if (attr.RequestType.GetTypeInfo().IsGenericType)
+                        {
+                            // remove `# from the generic type name
+                            var friendlyName = attr.RequestType.Name.Remove(attr.RequestType.Name.IndexOf('`'));
+                            // for generic, Schema will be TypeName[GenericTypeName]
+                            name = $"{friendlyName}[{string.Join(",", attr.RequestType.GetGenericArguments().Select(a => a.Name).ToList())}]";
+                        }
+                        else
+                            name = attr.RequestType.Name;
+                    }
+                    if(string.IsNullOrEmpty(name))
+                        continue;
 
-                    var name = parts.Last();
+                    var serializerSettings = SerializerSettings(attr.ContractResolver, attr.JsonConverter);
 
                     if (schemaRegistry.Definitions.ContainsKey(name))
                     {
                         var definitionToUpdate = schemaRegistry.Definitions[name];
-                        var serializerSettings = SerializerSettings(attr.ContractResolver, attr.JsonConverter);
-
                         definitionToUpdate.Example = FormatJson(provider, serializerSettings, false);
                     }
+                    else
+                    {
+                        // default schema not found, so put example directly to request
+                        request.Schema.Example = FormatJson(provider, serializerSettings, false);
```

Comment:
Oh, I had forgotten this property existed. I'm not sure if an OperationFilter such as this one should change that property, Perhaps it should be a SchemaFilter instead.